### DISCUSSION TITLE
Celebrating past TiddlyWiki contributors

### DISCUSSION THREAD

**Discussion by Jermolene** [TiddlyWiki/TiddlyWiki5#7983] [General]
A [chance conversation](https://github.com/TiddlyWiki/TiddlyWikiLinks/pull/58#issuecomment-1928118038) with @bimlas here on GitHub a few days ago prompted me to wonder if we can bring about a virtual reunion of past TiddlyWiki contributors.

Over the years, many people have become highly valued and trusted contributors to the core. For different personal and professional reasons, inevitably sometimes people move on to other things. (Occasionally, people come back after an extended absence with renewed enthusiasm (and skills) to bring to bear on TiddlyWiki).

This year marks the [20th anniversary](https://talk.tiddlywiki.org/t/tiddlywiki-is-19-years-old/8068) of the first release of TiddlyWiki which seems like a great opportunity to invite these past contributors to share some of their recollections.

In a comment below I'll tag people who have made contributions to the core in the past, but haven't been active here or in the forums in the last twelve months. Do please reply here if I've missed anyone.

If you are one of these contributors, I first of all would like to say a heartfelt thank you. From my own perspective, these twenty years have been like working with a sparkling, intertwined river of talented people. You have devoted your time and attention to something that I care about immensely, and I couldn't be more privileged than to have fallen into this position.

There are a few questions that might be fun if you felt like sharing.

* How are you doing?
* Has working on TiddlyWiki influenced your own way of working?
* Is TiddlyWiki reaching the potential you saw when you were contributing?
* Do you have any favourite memories of working on TiddlyWiki?

**Comment by Jermolene** [comment]
<details>
<summary>
First list of TiddlyWiki contributors
</summary>


* @xcazin
* @aelocson
* @tobibeer
* @araname
* @Skeeve
* @Sukima
* @andrey013
* @buggyj
* @rubaboo
* @idoine
* @mklauber
* @inmysocks
* @felixhayashi
* @EvidentlyCube
* @bimlas
* @gernert
* @welford
* @danielo515
* @Arlen22
* @ibnishak
* @natecain
* @ento
* @00SS
* @jbolila
* @Eucaly
* @roma0104
* @NicolasPetton
* @joebordes
* @FlashSystems
* @Evolena
* @Telumire
* @erwanm
* @fghhfg
* @Brennall
* @MaxGyver83
* @thediveo
* @nilslindemann
* @Spangenhelm
* @morosanuae
* @fu-sen
* @grayeul
* @SmilyOrg
* @Drakor
* @le-pako
* @davidjade
* @benwebber
* @OdinJorna
* @ng110

</details>

**Reply by Telumire** [reply]
### How are you doing?

My mental health wasn't in the best place last year, I wasn't able to focus / work, but I'm doing better now. I hope everyone is doing well !

### Has working on TiddlyWiki influenced your own way of working?

Yes ! I use it whenever I get the chance. I'm currently using it to gather my [research notes](https://designthinkerer.github.io/Scidelice/notes/notes%20TFE.html) for my bachelor thesis (it's a pwa aimed to help people to cook and eat well with science). I'm also using it to teach math (https://learning-math.tiddlyhost.com/), php (https://learning-php-mysql.tiddlyhost.com/), I created a [user guide for the steamdeck](https://steamdeckguide.tiddlyhost.com/), [a wiki](https://ifapme-dwfe.tiddlyhost.com/) for my classmates learning front end dev, and [many more](https://tiddlyhost.com/hub/user/Telumire). My biggest struggle with tiddlywiki is to complete a project before starting a new one, ahah.

### Is TiddlyWiki reaching the potential you saw when you were contributing?

The [latest annoncement](https://talk.tiddlywiki.org/t/announcing-the-multiwikiserver-plugin/9033) (multi user wiki) is exactly what I was missing in tiddlywiki, so yes, absolutely!

**Reply by grayeul** [reply]
I don't have a whole lot to add, and almost didn't reply -- but I figured I should at least take the time to say a very heartfelt thanks to @Jermolene for all his hard work on TiddlyWiki.  I had one very minor contribution, and while I kept wanting to get more time to help understand and document the behavior better, it never quite materialized.  @Jermolene made himself available for a one-on-one call and helped explain some of the codebase to me, which I really appreciated..... and I feel bad I never quite returned the favor by getting over the hump enough to really contribute.

While I don't necessarily use TiddlyWiki every day, I still use it quite a bit, and have convinced a few others of its benefit as well.  I have some useful info archived in a couple of TWs that I return to, and update periodically.   I have just recently retired from my day job, and now have a bit more time, and have started working on some other Open Source software.  I still hope that at some point I can get more familiar with the guts of TW -- but I'm grateful I have been able to come back to it over and over again as a user.  Thanks to all who helped make it what it is today!

**Comment by Jermolene** [comment]
<details>
<summary>
Second list of TiddlyWiki contributors
</summary>



* @ogoshima
* @clutterstack
* @telmiger
* @zahlman
* @maxthomax
* @tgrosinger
* @nome
* @edrex
* @FND
* @donmor
* @fkohrt
* @kixam
* @TiddlyVee
* @PauDeLuca
* @mpollio
* @eiro10
* @reflectionalist
* @spelufo
* @talha131
* @heroboy
* @rryan
* @ndarilek
* @cdruan
* @m42e
* @wincentbalin
* @AnthonyMuscio
* @FND
* @martinbudden
* @jdlrobson
* @markkerrigan
* @cdent
* @grmble
* @colmjude
* @PVHL
* @retostauss
* @BidiX

</details>

**Comment by Skeeve** [comment]
Hi Jeremy & thanks for the warm words.

I can't exactly remember, when I came across TW but it must have been around the time when TW5 just came out.

I can remember that I tried to use it in a project at work, where I had to put together a big questionnaire.

I also used it for personal documentation stuff while i was working on a project.

When using TW extensively I noticed a few limitations which I worked around by implementing some extensions for TW and so landed up in contributing a bit.

Unfortunately, for TW it is, not for me personally, my project and job completely changed in August 2019 and there was no real use anymore for TW in my work. I Just set up one small TW for noting down what to do if my colleague is on vacation and I have to take over a few of his tasks, but these are all not required anymore.

During Corona I couldn't follow one of my hobbies anymore: Dancing. Our dance classes were cancelled for a few months. But thanks to the owner of the dance school we got a lot of small video clips showing all the figures we learn in his classes. So the final project I did with TW5 was, to put together for my classmates a TW which lists all the clips related to our program, which I have on tiddlyhost. But unfortunately it wasn't used very much.

I have very fond memories of the time we had our Google Hangouts discussing some development with Jeremy and Mario. Unfortunately I think I won't be able to contribute any more because there are too many years of development I didn't follow.

I still can remember that I always wanted to change the scripting part of TW because I find it to inconsistent, but I never had good alternatives. I think I thought about implementing something like LISP (https://lisp-lang.org) because I felt the scripting already has some resemblance to that, but I never found the time to elaborate on that.

So Happy 20th anniversary to TW and congratulations to Jeremy and all the developers for this great program.

BTW. @Jermolene You remember my little son when we once met in London? He is the same age as TW, becoming 20 this year.

**Reply by sukima** [reply]
# How are you doing?

I'm doing well. It took a long time for me to fully utilize TiddlyWiki in my daily workflow but I've finally been using it daily for over 5 years consistently and I am so very glad it has been available for me.

Many of my co-workers obsess over tools like Obsidian, Notion, etc. but I keep chugging along with TiddlyWiki every happy and shaking my head at others continuing to search for that elusive productivity tool.

I think it helps that I have a very strong skill set in Web technology.

# Has working on TiddlyWiki influenced your own way of working?

It has, ever since I started making daily journal entries for work–usually just a list of ticket numbers and meetings attended–I am able to compile a year in review for my annual performance reviews. Because I settled on some consistent terminology in my tiddlers I was able to discover things like how many times I pair programmed or how many meetings I attended that year. Free form text search was very useful here.

In other projects I found that it was far easier and enjoyable to build documentation in disjointed ADHD like manor and have it all compiled into an interactive interface in a single HTML file. Being web based I could include all kinds of multimedia. I really enjoy that style of creativity. I am not a good organizer and TW lets me dump my brain and worry about organization after I've had a chance to marinate in my thoughts.

# Is TiddlyWiki reaching the potential you saw when you were contributing?

Yes and no. From a pure technical perspective I think TW5 we have today is the potential we saw decades ago. The web truly has become a wonderful environment and TW has captured that magic in a way many others have not.

However, from an adoption perspective I don't think it has. I also think that is not a reflection of TiddlyWiki but of the social shift we've seen these last 20 years. I remember early on that people would interact with technology in a way that augmented or enhanced how we thought. But more recently I've seen a massive shift from that tech-as-a-tool creative perspective into a more do-it-for-me instant gratification perspective.

For example, when I used to talk about TW and how it helped dump my brain I got engagement from others. Today when I do the same I'm quickly dismissed as an old fogy and the idea is lost on most. I also see this trend in more than just personal productivity but in creative works in general. The quality of craftsmanship has fallen very far and it scares me.

# Do you have any favourite memories of working on TiddlyWiki?

I remember working on my blog site in WordPress and falling in love with the idea of static site generators so I moved to Jekyll which use Markdown to make websites. After a bit I didn't like that it forced (at the time not as popular) a different runtime on to my development machine (Ruby) and any time I wanted to do something about JavaScript I kinda had to _fight the framework_. I looked for a more JS ecosystem solution and found Metelsmith. but like most things JavaScript that quickly became out of control and unmaintainable.

It was then that I realized the real issue wasn't the tech used but in how blogs were constructed. Well organized fully fleshed out thoughts. But what I wanted, what I needed was a dumping ground of whatever came out of my head. Snippets, tips, portfolios, links, and from time to time a fully thought out article.

TiddlyWiki fit this bill. I could focus on creating tiddlers. Have a one stop searchable, tag-able, transposable, and customizable website of all kinds of ideas. As a Single Page Application developer this scratched every itch I had.

Now I use it for all kinds of uses. [Minecraft tutorials](https://tritarget.org/wikis/minecraft/), daily task tracking at work, documentation, planning, and even end of life planning.

In one case it is my start page for work where I have tiddlers with `<audio>` tags for each music streaming service I like and I can pick which one to play while I work.

Though I've since stopped pinning up [posters](https://tiddlywiki.com/poster/) about TW in every coffee shop I visit I still use it to this day and love it more each time. I do hope that the ideas we cherished 20 years ago will come around and gain momentum again least I know that TiddlyWiki (even if it doesn't change) will fit me and my personality well up to and likely beyond my own passing.

Thank you very much to everyone who has and who will continue to make this philosophy realized through technology better and more attainable to everyone.

**Comment by mklauber** [comment]
### How are you doing?
Still doing well, and using TiddlyWiki regularly, though mostly for small personal projects.
### Has working on TiddlyWiki influenced your own way of working?
I definitely learned a lot working on tiddlywiki, especially since it was my first real introduction to doing javascript development.  However, my own professional work has moved in another direction, and I'm mostly doing sysadmin work now that doesn't overlap with Tiddlywiki very much. 
### Is TiddlyWiki reaching the potential you saw when you were contributing?
I do feel like TW is pretty amazing still, and regularly pull it out for different tasks.  It continues to be one of the simplest to setup wikis, and yet I feel like it also provides incredible flexibility that outshines most other similar tools.  It's still fantastic.  
### Do you have any favourite memories of working on TiddlyWiki?
I still fondly remember how many of us started to use TW for tabletop gaming around the same time.  It was a lot of fun, collaboratively writing plugins to roll dice, manage combats, and track all the things in the imaginary game world.  Seeing what other people wrote and being inspired to try using it as part of my game.

**Comment by FND** [comment]
TiddlyWiki had an immeasurable, enduring influence not just on my career, such as it is, but also on my values: To this day, I regularly find myself referring back to [fundamental concepts](https://prepitaph.org/articles/creative-privacy/) TiddlyWiki instilled in me - many of which are often forgotten or ignored elsewhere. Having this background thus helps me keep my bearings working in this industry, whether it's worshipping at the altar of technical complexity or even just remembering humans exist in the world of technology.

By TiddlyWiki, I mean people. It was an immense privilege interacting with and learning from this community and the group Jeremy built around it. It also helps reminding myself that this privilege was afforded to me by sheer happenstance; I hope to be paying it forward.

Every so often I get a little sad that I haven't managed to keep on top of the community's impressive development. However, I guess that's the natural order of things; passing the baton to the next generation and all. Jeremy also taught me there's value in being part of the diaspora.

Even so, I still have and use a number of TiddlyWiki documents (classic, mostly, though not exclusively) - and trust they will [remain operational](https://hachyderm.io/@FND/111874454670076867) longer than most files on my disk.

**Comment by kixam** [comment]
### How are you doing?

I am doing well thank you. Thank you for reaching out to me, to us :blush: 

### Has working on TiddlyWiki influenced your own way of working?

I started working on TiddlyWiki when i found this great Timeline plugin that had been discontinued and needed just a little work to be compatible with newer versions of TW5.

I continued working on other plugins and then even offered a tiny little bit change to the core to match my needs. I am mostly interested and focused on bringing more time management tools and features to TW5. I have been thinking on contributing again in the past few years, and share some of the tweaks that are sitting on my system waiting to be shared with the community.

I have been using TW in a professional context for almost 10 years now, and it helped a great deal with task management and as a personal knowledge base. TW also indirectly introduced me to Node.js and i got to learn to use that too in a broader manner.

### Is TiddlyWiki reaching the potential you saw when you were contributing?

I found out about TW after v5 was released. I was looking for a self contained HTML file that could work as a journal, to keep track of what I was working on when and for how long, keeping detailed information about every day and even every task, as well as summarizing it on larger scales such as months and years, and being able to filter and sort the data. So in a way, it was up to the potential i was looking for even as it was when i found about it.

Now i see TW as a mature tool, fit for professional use, and yet I have ideas for new plugins to add some features i would like to see there. As soon as I find the time to work on them, I will be sure to share these with the community.

### Do you have any favourite memories of working on TiddlyWiki?

I enjoyed working with some other members of the community, and in particular @felixhayashi as we adapted our respective plugins to improve performance and reduce footprint as we used to have several backend JS libraries in common (some were split since, and some were deprecated)

**Comment by telmiger** [comment]
How are you doing?

I’m fine, thanks. After improving TiddlyWiki for some years, I try to improve the "real world" now. I am active in local politics, ordered a heat pump and will build a photovoltaic installation on the flat roof of our house in March or April. There is time for physical activities and some culture now and then, our children have become adults and in my job I inform people about saving energy and greenhouse gases. By the way: My successful application and CV was a TiddlyWiki mainly.

    Has working on TiddlyWiki influenced your own way of working?

Yes (see above), and from the other side my work influenced my way of using TiddlyWiki. When I joined the TiddlyWiki community (Google Groups, March 2015) I had been looking for a note taking tool with superpowers. Steve Schneider and the SUNYpoly MOOC showed me what was possible then.

For my job in sales and consulting at that time, ToDoNow was very valuable (https://tid.li/tw5/tdn.html) – I had lots of tasks and managing deadlines and priorities was important. It was also very time consuming to build it, so I am not sure it was a time saver all in all ...

Nowadays I rely on taking notes for different activities in different wikis, all pimped with my own collection of plugins (https://tid.li/tw5/plugins.html). ToDo lists are not important any more.

Working on TiddlyWiki stuff helped me to stay in touch with HTML and CSS and motivated me to learn SVG.

GitHub on the other hand is something I did not take the time to learn properly, so I needed help from others and got frustrated about abandoned pull requests like the cats I tried to donate to the project (https://github.com/Jermolene/TiddlyWiki5/pull/2981).

    Is TiddlyWiki reaching the potential you saw when you were contributing?

When I started contributing I felt like a dwarf standing on the shoulders of giants only capable to throw in a thought here and there to keep the discussion alive. There were only few active members and some of them disappeared, so I was a bit afraid that TiddlyWiki would lose too much momentum. — That has changed completely. When I became a listener (I still follow some threads on https://talk.tiddlywiki.org/) I was sure that the community was healthier than I had ever seen it before and it grew and flourished even more since then. Congratulations to all of you, especially to @Jermolene for building and being such a great and helpful and inspiring team!

    Do you have any favourite memories of working on TiddlyWiki?

My first favourite "hack" was TextStretch. I did not expect it to survive for a long time, so I am really pleased to see it still works in my newest wikis, even if I never use it. https://tid.li/tw5/test/TWnews.html#TextStretch — Recently I was reminded of this when I stumbled upon http://www.kevinsteele.com/smackerel/black_white_00.html that has a similar effect.

One of the best features of TiddlyWiki is live hacking in an edit-text widget. This allows us to create tiny apps like my SVG Editor (https://tid.li/tw5/apps/svg.html) or to tune CSS tiddlers very efficiently. Just type and see the result immediately. Fantastic!

**Reply by xcazin** [reply]
Always loved TextStretch!

**Comment by zahlman** [comment]
I only briefly used TiddlyWiki for some project planning many years ago (and shelved the project), and wrote some documentation after figuring out how to do something relatively advanced. So I don't have the same perspective as other contributors here, but I do enjoy responding to these sorts of prompts.

More recently, I've been working (mentally) on a design for a system for collaborative writing and editing - something like a Markdown-based wiki, but with taggable, explicitly-connected "nodes" rather than pages related only by HTML hyperlinking. I think I can say TiddlyWiki inspired the idea that - with a web app - I could display multiple nodes simultaneously rather than attaching the page "wrapper" boilerplate to everything, and that I could have an option to organize the node divs into a "storyline" view as well as just a bunch of nodes connected with lines (like a stereotypical conspiracy theorist's bulletin board).

(I probably won't be working on this project any time for a while if ever; but I am planning to publish a more detailed design fairly soon.)

**Comment by xcazin** [comment]
Thank you Jeremy for the prompt! Many thanks as well to past and present developers, and to each active member of the community.

I have been contributing to the fr-FR translation since TW5 started, about 10 years ago, and my company sponsored the development of the quite ambitious [TWPUB](https://github.com/TWPUB/TWPUB-Tools), although on both fronts, I haven't been able to devote any serious time for several months. I also fancy that I may one day contribute two TW5 "books" back to the community: one about serving TW5 documents from your pocket —your mobile phone, that is— and another about building user interfaces for maintaining and querying smallish databases within TW5. Unfortunately, retirement is still a few years ahead!

I believe my fondest memories about working with TiddlyWiki were the few meetings we had with @jermolene and @jdjdjdjdjdjd, aka JD, around an early version of TWPUB, which my company is still using 5 years later. For many reasons, having the opportunity to collaborate with these two outstanding people together over 8 time zones on a project of mine was very striking.

Eventhough I didn't develop anything recently, I use TiddlyWiki every single day, whether to maintain our knowledge bases at work, or to add some notes to my personal wikis, or to test in a sandbox any wikitext construct that comes to mind out, of curiosity. All these wikis are served either by the NodeJS mode, or via WebDav thanks to the [recent work](https://saqimtiaz.github.io/tw5-plugins-sandbox/#WebDAV%20Utils) of @saqimtiaz. My best achievement so far as a TiddlyWiki power user has been to set up a wiki server on my wife's phone 5 years ago, which she  uses anytime to write down and categorise her thoughts about exhibitions, readings, or movies. Several hundred tiddlers later, she fully owns it, and never felt the need for one of those dedicated apps that may disappear any day ;-) 

In addition to witnessing every day the power of individual and collective intelligence, one of the great thing about Open Source efforts such as TiddlyWiki is that the solutions to a problem are often more ambitious and visionary than the time-constrained ad hoc solutions we typically see in industry. For example, the MultiWikiServer plugin [recently announced](https://talk.tiddlywiki.org/t/announcing-the-multiwikiserver-plugin/9033) by Jeremy  renewed my interest for TW once again as it promises to bring in new powerful features without altering the product nor making it more complex to use.

Since TiddlyWiki always seems to push the boundaries of expectations, it does influence my thoughts about what makes the Web truly useful nowadays: my fantasy about TW infinite possibilities is what drives my desire of (net)working differently. For instance, I was frustrated at work by the fact that the main ebook format was EPUB, no more than a static collection of HTML pages. For 20 years, I had instead wanted fully interactive web books and it seemed obvious that TiddlyWiki was the only format that could possibly fulfill this role. This lead me to sponsor the development of an EPUB ehancer based on TiddlyWiki, aka TWPUB. Once it is complete, il will allow us to automatically convert the EPUBs that our publishing customers entrust to us into true read-write, interactive and multimedia web books that final customers can enjoy as they enjoy rich websites today.

**Comment by bimlas** [comment]
Thank you @Jermolene for thinking of us. :) Personally, it is always a great pleasure to use and develop TiddlyWiki, but I like the community built around it even more: lots of smart, broad-minded people who solve their own tasks in incredibly diverse ways and this gives inspiration and encouragement to others. Whatever direction my note-taking habit or even my programming career takes, I will always think of the TiddlyWiki era with gratitude and happiness.

> How are you doing?

Thank you, luckily my life is going well. My relationship didn't turn out the way I wanted, but I'm very happy that my ex-partner and I found each other, because I think we raised each other to a higher level. My professional life is also moving in the right direction, I am finally a full-fledged devops / programmer in a place where I can learn a lot and where I can ask my questions (before that I was the only developer at a small company, which has its own beauty, but the there were many obstacles to professional development).

> Has working on TiddlyWiki influenced your own way of working?

I think the answer is yes, because in addition to TiddlyWiki's programming solutions, I was also influenced by its philosophy: all data should be as minimal and general as possible, and special solutions should only be used in very justified cases. When I get a task in which I have to structure data, I try to use the solutions I learned in TiddlyWiki to figure out how I would organize them if the data were tiddlers.

> Is TiddlyWiki reaching the potential you saw when you were contributing?

Definitely yes. I remember the time when I was developing the https://bimlas.github.io/demo/tw5/property-comparison.html solution and that made me aware of the true nature of TiddlyWiki: it is a framework for objects. In fact, each tiddler is an object that has some exceptionally handled properties (title, text, tags), but it can easily handle a music collection, family tree, schedule, anything in which objects have properties and a relationship between them needs to be represented. Filters, macros and widgets are so well abstracted that they can be used for everything due to their generality. I really like that TiddlyWiki is so flexible and that note management is really just one example of how it can be used, but it's best used as a framework.

I would add that wiki syntax is not always easy to work with, but it is a necessary part of flexibility. :)

> Do you have any favourite memories of working on TiddlyWiki?

One of these is the above-mentioned relevation, but in reality there is no such specific moment. I like when we come up with solutions together with the community, but I also find joy when I experience the sublime moment of reaching a solution alone.

I wish you much success and happiness in your personal life as well, and thank you for giving the community the energy, knowledge and wisdom you have invested in TiddlyWiki! This thanks goes to all contributors as well! :clap:

**Comment by donmor** [comment]
Hello there! BTW I finally made Tiddloid 2.5.0 last week (after an absence of a year _(:зゝ∠)_)

**Comment by Jermolene** [comment]
A big thank you for the responses, which I have very much enjoyed and much appreciate. It is great to hear from everyone.

@Telumire – I'm glad to hear that things are improving, and I hope your upward trajectory continues apace. Very pleased to hear that TiddlyWiki is still part of your life.

@grayeul - delighted to hear you're still enjoying TiddlyWiki, and I hope you'll continue. Absolutely no problem about your contributions not materialising, it was still a useful learning experience for me.

@Skeeve - I hope your son is thriving, I remember meeting you both very well. Your contributions definitely helped stretch the core, and are still much appreciated.

@Sukima - thanks for your thoughtful reflections, which remind me how much I enjoyed your perspective. I'm glad you are still on the same journey.

@mklauber - great to hear from you. You made a big impact very swiftly, I think really helping to coalesce a community that was latent and inchoate. I still feel an enormous kinship for the rpg mindset and consider it a mark of honour to have made an impact in that space.

@FND - thank you – as you know, I admire your writing enormously, and enjoyed [your recent post about browser file system access](https://prepitaph.org/articles/web-fs/). It is very gratifying that you credit TiddlyWiki, and I much appreciate it.

@kixam – thank you for the thoughtful comments, I like the idea that TiddlyWiki's most important and useful characteristics have been there from the beginning.

@telmiger - great to hear of your journey, and I admire the environmental work that you are doing tremendously. I'll have a look at #2981 again.

@zahlman - that sounds interesting. Maybe TiddlyMap could help you prototype your ideas?

@xcazin - thank you, great to hear your perspective. I also enjoyed working with @jdjdjdjdjd. The TWPub work was very productive from my point of view, and will influence the core for some time to come as we integrate the navigation improvements to the core. I am also delighted that we got the basic EPUB-to-tiddlers conversion going so well, it makes TWPub useful for a wide audience.

@bimlas - thank you for the kind words and the interesting reflections, the data model of TiddlyWiki is something we take for granted but its success at modelling a wide range of data deserves wider analysis. 

@donmor - good to see you back. I know a lot of people depend on Tiddloid, so it's great to have an update.

By the way, this response is not intended to close this discussion. I'd love to hear from more contributors, and continue this discussion.

**Comment by morosanuae** [comment]
**How are you doing?**
I'm fine, thank you! I'm still alive! Maybe even more alive after I've abandoned the city and the IT industry for good and I've moved to the country side with my family. I don't want to be part of this IT thing anymore because I don't like the direction is heading. Technology has become a scope in itself not just a mean to achieve something really useful. It's moving very fast towards nowhere, exactly like our modern civilization! And I don't want to hear anything about AI also. This bulls**t will replace almost all humans soon. I hate that and I believe this is very wrong! I'm telling you honestly what I think, I don't want to offend somebody.

**Has working on TiddlyWiki influenced your own way of working?**
I've used TiddlyWiki only as a personal digital brain to store my knowledge, any kind of knowledge, and I'm still using it today for the same purpose. I had to accommodate it's vision at first but was for a greater good in the end.

**Is TiddlyWiki reaching the potential you saw when you were contributing?**
I haven't discovered its real potential at first. It seemed to me like a simplistic app and I've discarded it fairly easily. Then, somehow I've discovered WikiText and had a moment of "Eureka". It has so much power with WikiText. But I still don't like the way the documentation is structured and organized. For me it's really a mess. I really can't find what I'm looking for in a short amount of time. I don't know, maybe it's just my problem.

**Do you have any favourite memories of working on TiddlyWiki?**
I've enjoyed very much developing in TiddlyWiki, finding solutions and be creative. The fact that TiddlyWiki is highly customisable by an average user through WikiText is a major advantage compared to other knowledge management apps. On the other hand I'm glad that I've offered something in return by contributing to the core (mainly the Explorer tab) and to the community with TW-Icons site. It is a very good thing that TiddlyWiki is evolving continuously but it's becoming very difficult for me to keep track of all the new developments. So I don't think I will contribute anymore. But the "What's New" page is always a pleasure to read when a new version is out.

In conclusion long live TiddlyWiki and its community!

**Comment by inmysocks** [comment]
How are you doing?
Not dead yet. Also busy. My previous project ended unexpectedly and setting up what comes next has been taking a lot of my energy so I have had to be careful about which projects I am spending time on.
For those that are interested, here is a quick rundown:
- continuing work on the house. It was an empty building and we have been doing all of the work ourselves (electrical, plumbing, carpentry, some custom home automation, etc.)
- Some art projects that have not yet produced anything sharable
- A lot of what I am going to claim is professional development, but is mostly me learning more coding by creating stuff on one will ever use
- Releasing a simple iOS game in the app store to see if I could do it
- lots of gardening, fighting a war against snails and loosing
- Using IR photography as part of a local archeology project
- Finally setting up a wireless mesh network using the research I did in grad school

The current 'useful' project is working on a project to pitch to the US department of agriculture for helping handle avian flu outbreaks.

Has working on TiddlyWiki influenced your own way of working?
Yes. It is still the only tool I have found that I will actually use for organizing notes and documenting what I do. I am still working on the database project that uses tiddlywiki style filters for data retrieval and connecting that to Bob as a storage back-end.
Almost all of the projects I work on end up having a modular design similar to tiddlywiki.

Is TiddlyWiki reaching the potential you saw when you were contributing?
Tiddlywiki has always been very flexible, and that certainly hasn't changed. The places where what I hoped to see hasn't happened are mostly in things that I was creating myself, like federation. I am still looking at ways to make the initial setup and saving simpler for everyone, but other than my very slow work with Bob not much has come from it.
One big thing that I would like to see more done with is code optimization, the biggest example is tag handling. Tagging is a very important part of tiddlywiki and I would like to be able to have a complex wiki that has 10000 tags without trouble. I had some vague plans for using a B+ tree structure for handling tags which would prevent the slowdowns caused by using indexOf to test for the existence of a tag. (I haven't kept up with the state of the code for a while so this may have been addressed already.)

Do you have any favourite memories of working on TiddlyWiki?
Meeting everyone in Oxford and Driving around my little robot that ran as a tiddlywiki plugin are certainly fond memories.

Edit: Writing the simple interactive fiction engine in tiddlywiki was also a lot of fun.

**Comment by welford** [comment]
I don't have much to add to this other than a quick thank you to @Jermolene for Tiddlywiki - I use it daily, and a short answer to this question:

> Has working on TiddlyWiki influenced your own way of working?

Yes, Tiddlywiki had influenced how i work - and the work i have wanted to do influenced my (very minor) contributions to the project. 

At a company I used to work for I kept notes in the original tiddlwiki for many years, someone else in that company had setup a nice little .hta webpage as a quick way of deploying simple tools to non programmers. Tiddlywiki 5 came out just after i left that company, and around that time I had the lightbulb moment of combining these two resulting in my https://github.com/welford/twexe plugin. 

Being able to write batch/cmd scripts directly in tiddlywiki then run those script from buttons embedded in other tiddlers that themselves document how they should be used is, in my opinion, a very powerful concept. I now have a lot of handy batch files and documentations (including a button that will export tagged tiddlers to a blog) all bundled up in one 10 year old wiki.

> Is TiddlyWiki reaching the potential you saw when you were contributing?

Personally yes - though I think some aspects of it could be simpler. I always find myself going back to the docs when i need to write new filters, something about the syntax doesn't stick with me.
